src/HttpClient.php

```diff
@@ -121,13 +122,17 @@ public function request(string $verb, string $route, $body = [], $headers = []):
      */
     protected function createRequest(string $verb, string $route, $body = [], $headers = [])
     {
-        $url = Router::getRouteUrl($route);
-
-        if (strtoupper($verb) === 'GET') {
-            $url .= '?' . http_build_query($body);
+        $verb = strtoupper($verb);
+        $url = new Uri(Router::getRouteUrl($route));
+
+        if (in_array($verb, ['POST', 'PUT', 'PATCH'])) {
+            $body = json_encode($body, JSON_THROW_ON_ERROR);
+        } else {
+            $url = $url->withQuery(http_build_query($body));
+            $body = null;
         }
```

Comment:
already approved as this is a non blocking comment, only a suggestion to clean it up for readability:

maybe avoid using if else, e.g. swap to private methods with early returns or however you want to do it

Reply:
That's an idea! I do enjoy early returns. I'll experiment a little and see if I can make things a bit cleaner. 👍

Reply:
Ended up deciding it wouldn't be any simpler with a separate method, since it needs to change both the `$body` and potentially the `$url` and that would mean fussing around with multiple return values.

Doable, of course, but not exactly something PHP enjoys or handles very well.